src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs

```diff
@@ -35,30 +36,25 @@ internal sealed class SynthesizedRecordEquals : SynthesizedInstanceMethodSymbol
             MethodSymbol? otherEqualsMethod,
             int memberOffset)
         {
+            // If the parameter is a struct type, it should be declared `in`
+            // and we need to call EnsureIsReadOnlyAttributeExists().
+            Debug.Assert(!parameterType.IsStructType());
+
             var compilation = containingType.DeclaringCompilation;
-            bool isStruct = parameterType.IsStructType();
 
             _equalityContract = equalityContract;
             _otherEqualsMethod = otherEqualsMethod;
             _memberOffset = memberOffset;
 
             ContainingType = containingType;
+            IsVirtual = !isOverride;
             IsOverride = isOverride;
             Parameters = ImmutableArray.Create(SynthesizedParameterSymbol.Create(
                 this,
-                TypeWithAnnotations.Create(parameterType, nullableAnnotation: isStruct ? NullableAnnotation.NotAnnotated : NullableAnnotation.Annotated),
+                TypeWithAnnotations.Create(parameterType, nullableAnnotation: NullableAnnotation.Annotated),
                 ordinal: 0,
-                isStruct ? RefKind.In : RefKind.None));
+                RefKind.None));
             ReturnTypeWithAnnotations = TypeWithAnnotations.Create(compilation.GetSpecialType(SpecialType.System_Boolean));
```

Comment:
>compilation.GetSpecialType(SpecialType.System_Boolean) [](start = 67, length = 54)

We should use static helper from Binder, which reports use-site diagnostics.

Reply:
Logged https://github.com/dotnet/roslyn/issues/44915.